packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -131,6 +131,13 @@ class ColdRunner extends ResidentRunner {
       await connectToServiceProtocol();
     } catch (error) {
       printError('Error connecting to the service protocol: $error');
+      // https://github.com/flutter/flutter/issues/33050
+      // TODO(blasten): Remove this check once https://issuetracker.google.com/issues/132325318 has been fixed.
+      if (await hasDeviceRunningAndroidQ(flutterDevices) &&
+          error.toString().contains(kAndroidQHttpConnectionClosedExp)) {
+        printStatus('🔨 If you are using an emulator running Android Q Beta, consider using an emulator running API level 29 or lower.');
```

Comment:
Should this have said 28?

(it might not be worth changing now, I just came across it because a user reported what might be this issue at https://github.com/Dart-Code/Dart-Code/issues/2061 and the error confused me.. currently all the API levels are 29 or lower)